### DISCUSSION TITLE
No compiler error for not yet implementing abstract template class properties

### DISCUSSION THREAD

**Discussion by osamakawish** [dotnet/roslyn#57486] [Q&A] [answered]
Not sure if this should be marked as an issue, but if it is, I'll turn it into one if necessary. Also not sure if this belongs here or the wpf one.

Anyways, I've got a Windows 10 and am making a .Net 5.0 WPF app.

I've got a class like this:

```c#
public abstract class AnimatablePropertyControl<TValue> : System.Windows.UIElement
    {
        public abstract BitmapImage Icon { get; protected set; }
        public abstract TValue Value { get; protected set; }
    }
```

And another class as follows:
```c#
public partial class DoublePropertyControl : AnimatablePropertyControl<double>
    {
         ...
    }
```

Problem is, `DoublePropertyControl` isn't giving me any errors for not defining the `Icon` or `Value` properties. Visual Studio isn't offering me any "Generate Overrides" for this either. I tested if it works fine when I don't have a template class by just tossing in code like:

```c#
public abstract class X
{
    public abstract int Fun { get; set; }
}

public class Y : X
{

}
```

And Visual Studio does point out an error and the context menu does offer me the option to implement the abstract class in the code above.

So yeah. For some reason, the same stuff doesn't pop up when I implement an abstract template class.

**Comment by osamakawish** [comment, marked as answer]
Never mind. Issue comes back up after closing and reopening Visual Studio lol.

**Reply by jasonmalinowski** [reply]
@osamakawish If you see this is again, file a bug. Close/reopen _shouldn't_ be fixing things!
